File: src/main/java/com/koddy/server/global/exception/OAuthExceptionResponse.java
package com.koddy.server.global.exception;

import com.koddy.server.auth.domain.model.oauth.OAuthUserResponse;

public record OAuthExceptionResponse(
        String id,
        String name,
        String email,
        String profileImageUrl
) {
    public OAuthExceptionResponse(final OAuthUserResponse oAuthUserResponse) {
        this(
                oAuthUserResponse.id(),
                oAuthUserResponse.name(),
                oAuthUserResponse.email(),
                oAuthUserResponse.profileImageUrl()
        );
    }
}

File: src/main/java/com/koddy/server/global/exception/OAuthExceptionResponse.kt
package com.koddy.server.global.exception

import com.koddy.server.auth.domain.model.oauth.OAuthUserResponse

data class OAuthExceptionResponse(
    val id: String,
    val name: String,
    val email: String,
    val profileImageUrl: String,
) {
    constructor(oAuthUserResponse: OAuthUserResponse) : this(
        id = oAuthUserResponse.id(),
        name = oAuthUserResponse.name(),
        email = oAuthUserResponse.email(),
        profileImageUrl = oAuthUserResponse.profileImageUrl(),
    )
}
